File: frontend/src/app/core/services/response/room-detail.response.ts
import { RoomStatus } from '../../models/room-status.enum';
import { ParticipantResponse } from './participant.response';
import { UserResponse } from './user.response';

export interface RoomDetailResponse {
  id: number;
  name: string;
  description?: string;
  status: RoomStatus;
  createdBy: UserResponse;
  participants: ParticipantResponse[];
  createdAt: string;
}

File: backend/src/main/java/com/finki/backend/web/response/RoomDetailResponse.java
package com.finki.backend.web.response;

import com.finki.backend.core.domain.enums.RoomStatus;
import lombok.Builder;

import java.time.Instant;
import java.util.List;

@Builder
public record RoomDetailResponse(
        Long id,
        String name,
        String description,
        RoomStatus status,
        UserResponse createdBy,
        List<ParticipantResponse> participants,
        Instant createdAt
) {
}
